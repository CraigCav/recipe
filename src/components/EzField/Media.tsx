import React, {useRef, useState, useEffect} from 'react';
import {createPortal} from 'react-dom';
import {CacheProvider} from '@emotion/core';
// eslint-disable-next-line
import createCache from '@emotion/cache';
import {EzGlobalStyles} from '../index';

const sizes = {
  small: {width: 750, height: 1334}, // phone e.g. iphone 6
  medium: {width: 1280, height: 800}, // tablet e.g. amazon fire hd
  large: {width: 1920, height: 1080}, // laptop e.g. dell xps 13
};

const IFrameContent = ({iframeEl, children}) => {
  const cache = useRef(createCache({container: iframeEl.ownerDocument.head, key: 'frame'}));

  // Remove any injected stylesheets from the page when the component is unmounted
  React.useEffect(() => () => cache.current.sheet.flush());

  return (
    <>{createPortal(<CacheProvider value={cache.current}>{children}</CacheProvider>, iframeEl)}</>
  );
};

const Media = ({size, children}) => {
  const iframeEl = useRef(null);
  const linkRef = useRef(null);
  const [src, setSrc] = useState(undefined);

  const [container, setContainer] = useState(null);

  useEffect(() => {
    setContainer(iframeEl.current);
  }, [container]);

  return (
    <iframe
      title={size}
      frameBorder="0"
      allowFullScreen
      ref={iframeEl}
      style={{border: 'none', margin: 0, ...sizes[size]}}
      scrolling="no"
      src={src}
    >
      {iframeEl.current && !src && (
        <IFrameContent iframeEl={iframeEl.current}>
          <link
            href="https://fonts.googleapis.com/css?family=Montserrat:400,400i,700,700i&display=swap"
            rel="stylesheet"
            ref={linkRef}
          />
          <EzGlobalStyles />
          {children}
          {linkRef.current &&
            setSrc(
              `data:text/html,${encodeURIComponent(
                `<html>${linkRef.current.ownerDocument.head.outerHTML}<body>${iframeEl.current.innerHTML}</body></html>`
              )}`
            )}
        </IFrameContent>
      )}
    </iframe>
  );
};

export default Media;
