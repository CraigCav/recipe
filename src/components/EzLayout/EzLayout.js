import PropTypes from 'prop-types';
import styled from 'react-emotion';
import {base, layout, spacing} from './EzLayout.styles';
import {standard} from '../../themes';

/**
 * Layout provide common ways to arrange content in a single horizontal row.
 */
const EzLayout = styled('div')(base, layout, spacing);

EzLayout.propTypes = {
  /**
   * Controls the arrangement of items within the layout.
   */
  layout: PropTypes.oneOf(['basic', 'right', 'equal', 'split']),
  /**
   * The theme controlling the default styles.
   */
  theme: PropTypes.shape({
    colors: PropTypes.object,
  }),
};

/**
 * defaultProps
 * @property {string} layout - uses the basic layout by default.
 * @property {object} theme - uses the standard theme by default.
 */
EzLayout.defaultProps = {
  layout: 'basic',
  theme: standard,
};

/**
 * @component
 */
export default EzLayout;
