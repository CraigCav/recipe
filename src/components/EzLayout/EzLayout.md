---
name: EzLayout
title: Layout
path: "/components/ez-layout"
---

The layout component provide common ways to arrange the content of a screen. The layout component controls the horizontal spacing of its content.

---

## Best Practices

Layouts should:

* Be used for small-scale layout tasks when you want to lay out a horizontal row of content

Layouts should not:

* Be used for large-scale page layout. For large-scale layouts, consider using a [Card](./ez-card) within [Page Content](./ez-page-content)
* Wrap content onto multiple lines. The content inside a layout should be designed as not to exceed the available horizontal space.

---

## Examples

### Basic layout

The `basic` layout arranges content into a single row with even spacing between the content items.

```jsx
<EzCard
  className={css(`
  .basic > * {
    background-color: yellow;
  }
`)}
>
  <EzLayout layout="basic" className="basic">
    <div>Content</div>
    <div>Content</div>
    <div>Content</div>
    <div>Content</div>
  </EzLayout>
</EzCard>
```

### Right layout

The `right` layout arranges content into a single row, aligned to the right edge of its container, with even spacing between the content items. The `right` layout can be used, for example, for aligning action buttons within a card.

```jsx
<EzCard
  className={css(`
  .right > * {
    background-color: coral;
  }
`)}
>
  <EzLayout layout="right" className="right">
    <div>Content</div>
    <div>Content</div>
    <div>Content</div>
    <div>Content</div>
  </EzLayout>
</EzCard>
```

### Equal layout

The `equal` layout arranges content into equal widths filling a single row. The `equal` layout can be used, for example, for evenly spacing spending statistics across a single row.

```jsx
<EzCard
  className={css(`
  .equal > * {
    background-color: aqua;
  }
`)}
>
  <EzCardSection>
    <EzLayout layout="equal" className="equal">
      <div>Equal Layout</div>
      <div>Content</div>
      <div>Content</div>
    </EzLayout>
  </EzCardSection>
  <EzCardSection>
    <EzLayout layout="equal" className="equal">
      <div>Equal Layout</div>
      <div>Content</div>
      <div>Content</div>
      <div>Content</div>
    </EzLayout>
  </EzCardSection>
</EzCard>
```

### Split layout

The `split` layout arranges content into two groups on a single row, one aligned to the left edge of its container with the other to the right edge. The `split` layout will usually be [combined with other layout arrangements inside](#combining-layouts). The `split` layout can be used, for example, for aligning secondary buttons for next steps to the bottom left of a card, while aligning primary calls-to-action to the bottom right of a card.

```jsx
<EzCard
  className={css(`
  .split > * {
    background-color: palegreen;
  }
`)}
>
  <EzLayout layout="split" className="split">
    <div>Split Layout</div>
    <div>Content</div>
  </EzLayout>
</EzCard>
```

### Combining Layouts

The layout component can be nested in order to provide more complex or unique arrangements of components. For example, a toolbar might use a basic layout inside of an equal layout, to arrange logical groupings of action buttons into visual groups with even spacing between the grouped buttons.

```jsx
<EzCard
  className={css(`
  .basic > * {
    background-color: yellow;
  }
  .equal > * {
    background-color: aqua;
  }
`)}
>
  <EzCardSection>
    <EzLayout layout="split">
      <EzLayout layout="basic" className="basic">
        <div>Basic Layouts in a Split Layout</div>
        <div>Content</div>
      </EzLayout>
      <EzLayout layout="basic" className="basic">
        <div>Content</div>
        <div>Content</div>
      </EzLayout>
    </EzLayout>
  </EzCardSection>
  <EzCardSection>
    <EzLayout layout="split">
      <EzLayout layout="equal" className="equal">
        <div>Equal Layouts in a Split Layout</div>
        <div>Content</div>
      </EzLayout>
      <EzLayout layout="equal" className="equal">
        <div>Content</div>
        <div>Content</div>
      </EzLayout>
    </EzLayout>
  </EzCardSection>
</EzCard>
```

---

## Limitations

* Layout does not currently support vertically stacked content
* Layout does not currently support varying horizontal/vertical stacked layouts across breakpoints.

---

## Related components

* [Card](./ez-card)
* [Page Content](./ez-page-content)
