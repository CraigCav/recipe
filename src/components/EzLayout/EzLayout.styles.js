import {css} from 'react-emotion';
import {margin} from 'polished';
import {responsive} from '../../styles';

export const base = css`
  display: flex;
`;

const reset = css`
  align-items: normal;
  flex-direction: row;
  justify-content: normal;

  > * {
    flex-basis: auto;
    flex-grow: 0;
  }
`;

export const basic = () => css`
  align-items: center;
  justify-content: flex-start;
`;

export const right = css`
  align-items: center;
  justify-content: flex-end;
`;

export const equal = () => css`
  align-items: center;
  justify-content: space-between;

  > * {
    flex-basis: 0;
    flex-grow: 1;
  }
`;

export const split = () => css`
  align-items: center;
  justify-content: space-between;
`;

export const stack = css`
  flex-direction: column;
`;

export const layout = responsive('layout', {
  basic,
  right,
  equal,
  split,
  stack,
  reset,
});

const setSpacing = (top, left) => css`
  > *:not(:first-child) {
    ${margin(top, null, null, left)};
  }
`;
const leftSpacing = ({theme}) => setSpacing(null, theme.spacing.sm);
const topSpacing = ({theme}) => setSpacing(theme.spacing.sm, null);

export const spacing = responsive('layout', {
  basic: leftSpacing,
  right: leftSpacing,
  equal: leftSpacing,
  stack: topSpacing,
  reset: setSpacing(0, 0),
});
