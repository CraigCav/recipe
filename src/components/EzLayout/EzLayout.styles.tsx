import {css} from '@emotion/core';
import {responsive} from '../../styles';

export const base = () => css`
  display: flex;
`;

const reset = () => css`
  align-items: normal;
  flex-direction: row;
  justify-content: normal;
  flex-wrap: nowrap;

  > * {
    flex-basis: auto;
    flex-grow: 0;
    width: auto;
  }
`;

export const basic = () => css`
  align-items: center;
  justify-content: flex-start;
`;

export const right = () => css`
  align-items: center;
  justify-content: flex-end;
`;

export const equal = () => css`
  align-items: center;
  justify-content: space-between;

  > * {
    flex-basis: 0;
    flex-grow: 1;
  }
`;

export const split = () => css`
  align-items: center;
  justify-content: space-between;
`;

export const stack = () => css`
  flex-direction: column;
`;

const columnWidth = ({columns, theme}) => {
  const width = n => ({width: `calc(${n < 2 ? 100 : 100 / n}% - ${theme.spacing.sm})`});

  if (typeof columns !== 'object') return width(columns);

  return responsive(
    'columns',
    Object.keys(columns).reduce((res, key) => {
      res[columns[key]] = width(columns[key]);
      return res;
    }, {})
  )({columns, theme});
};

const tile = ({columns, theme}) => css`
  flex-wrap: wrap;

  /* multiply by -1 to negate the margin, since the container absorbs the additional outer margin of the children) */
  margin: calc(${theme.spacing.sm} / 2 * -1);

  > * {
    flex-basis: auto;
    flex-grow: 0;
    ${columnWidth({columns, theme})};
    /* ↓ half the value, to avoid doubling up the space between columns */
    margin: calc(${theme.spacing.sm} / 2);
  }
`;

const cluster = ({theme}) => css`
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;

  /* multiply by -1 to negate the margin, since the container absorbs the additional outer margin of the children) */
  margin: calc(${theme.spacing.sm} / 2 * -1);

  > * {
    /* ↓ half the value, to avoid doubling up the space between columns */
    margin: calc(${theme.spacing.sm} / 2);
  }
`;

export const layout = responsive('layout', {
  basic,
  right,
  equal,
  split,
  stack,
  tile,
  cluster,
  reset,
});

const setSpacing = (top, left) => css`
  && > * + * {
    margin-top: ${top};
    margin-left: ${left};
  }
`;
const leftSpacing = ({theme}) => setSpacing(null, theme.spacing.sm);
const topSpacing = ({theme}) => setSpacing(theme.spacing.sm, null);

export const spacing = responsive('layout', {
  basic: leftSpacing,
  right: leftSpacing,
  equal: leftSpacing,
  stack: topSpacing,
  reset: () => ({'&& > * + *': {marginLeft: 0, marginRight: 0, marginTop: 0, marginBottom: 0}}),
});

export const wrapper = () => css`
  /* ↓ Suppress horizontal scrolling caused by the negative margin in some circumstances */
  overflow: hidden;
`;
