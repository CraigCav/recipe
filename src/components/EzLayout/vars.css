:root {
  --recipe-layout-gap: var(--recipe-global-static-size-150);
}