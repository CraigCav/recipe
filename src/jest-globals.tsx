import {shallow as enzymeShallow, render as enzymeRender, mount as enzymeMount} from 'enzyme';
import React from 'react';
import {create as reactCreate} from 'react-test-renderer';
import {render as rtlRender, cleanup} from 'react-testing-library';
import {renderToStaticMarkup} from 'react-dom/server';
import {ThemeProvider} from 'emotion-theming';
import {standard} from './themes';

import '@testing-library/jest-dom/extend-expect';

const renderWithTheme = renderFn => (component, ...rest) =>
  renderFn(<ThemeProvider theme={standard}>{component}</ThemeProvider>, rest);

export const shallow = tree => {
  const wrapper = enzymeShallow(<ThemeProvider theme={standard} />);
  const instance = wrapper.instance() as any;
  const context = instance.getChildContext();

  return enzymeShallow(tree, {context});
};

export const mount = tree => {
  const wrapper = enzymeShallow(<ThemeProvider theme={standard} />);
  const instance = wrapper.instance() as any;
  const context = instance.getChildContext();

  return enzymeMount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

export const renderToHtml = renderWithTheme(renderToStaticMarkup);

export const render = renderWithTheme(enzymeRender);

export const fullRender = renderWithTheme(rtlRender);

export const create = renderWithTheme(reactCreate);

afterEach(cleanup);
