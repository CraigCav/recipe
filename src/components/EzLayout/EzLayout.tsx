/** @jsx jsx */
import {jsx} from '@emotion/core';
import styled from '@emotion/styled';
import {base, layout, spacing, wrapper, alignX, alignY} from './EzLayout.styles';
import {breakpoints} from '../../themes/standard';

/**
 * Union of properties from T
 */
type Properties<T> = {[K in keyof T]: Pick<T, K>}[keyof T];

/**
 * Require at least one of the properties of T
 */
type RequireAtLeastOne<T> = Required<Properties<T>>;

/**
 * Create a type that disallows values for any of the keys of T
 */
type Without<T> = {[P in keyof T]?: never};

type LayoutTypes = 'basic' | 'right' | 'equal' | 'split' | 'stack' | 'tile' | 'cluster';
type Sizes = keyof typeof breakpoints;
type AlignX = 'left' | 'right' | 'center';
type AlignY = 'top' | 'bottom' | 'center' | 'stretch';

/**
 * A type that describes how the layout will adapt to different breakpoints
  @example
  ```
  const options: Responsive<LayoutTypes>  = {
    base: 'stack';
    medium: 'split';
    large: 'tile';
  };
  ```
 */
type Responsive<T> = Partial<Record<'base' | Sizes, T>> & RequireAtLeastOne<Record<Sizes, T>>;

/**
 * The layout prop of the EzLayout component. Accepts either a simple key (one of the LayoutTypes),
 * or a Responsive<LayoutTypes> object describing how the layout will adapt to different
 * breakpoints.
  @example
  ```
  const options: Responsive<LayoutTypes>  = {
    base: 'stack';
    medium: 'split';
    large: 'tile';
  };
  ```
 */
type Layout<T, Set = T> = {
  layout: T | (Responsive<Set> & RequireAtLeastOne<Responsive<T>>);
  alignX?: AlignX | (Responsive<AlignX> & RequireAtLeastOne<Responsive<AlignX>>);
  alignY?: AlignY | (Responsive<AlignY> & RequireAtLeastOne<Responsive<AlignY>>);
};

/**
 *  The required props for the LayoutTypes["tile"] layout variation.
 */
type TileProps = {columns: number | Responsive<number>};

type Props =
  | (TileProps & Layout<'tile', LayoutTypes>)
  | (Partial<Layout<Exclude<LayoutTypes, 'tile'>>> & Without<TileProps>);

/**
 * Layout provide common ways to arrange content in a single horizontal row.
 */
const InnerLayout = styled.div<Props>(base, layout, spacing, alignX, alignY);
const Wrapper = styled.div<any>(wrapper);

const requiresNegativeMargin = layoutProp =>
  (typeof layoutProp === 'object' && 'tile' in layoutProp) || layoutProp === 'tile';

const EzLayout: React.FC<Props> = props => {
  if (!requiresNegativeMargin(props.layout)) return <InnerLayout {...props} />;

  // Note: The layout component needs to the respect white space that might be applied by a parent layout component.
  // A wrapper element is included here to insulate content from the applied negative margin.
  return (
    <Wrapper>
      <InnerLayout {...props} />
    </Wrapper>
  );
};

/**
 * defaultProps
 * @property {string} layout - uses the basic layout by default.
 * @property {object} theme - uses the standard theme by default.
 */
EzLayout.defaultProps = {
  layout: 'basic',
};

export default EzLayout;
