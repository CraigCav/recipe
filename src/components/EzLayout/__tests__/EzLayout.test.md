### base styles should be reset when applying breakpoint styles

Note: the black background here is to highlight the alignment of the items.
When items are misaligned, the content items aren't symmetric.

```jsx
<div>
  <Global
    styles={css(`
    .basic > * > * {
      background-color: yellow;
    }
  `)}
  />
  <EzCard>
    <EzLayout
      layout={{base: 'stack', medium: 'split'}}
      className="basic"
      style={{backgroundColor: 'black'}}
    >
      <div>Content</div>
      <div>Content</div>
    </EzLayout>
  </EzCard>
</div>
```

### Collapse to single column on small screens

```jsx
<div>
  <Global
    styles={css(`
    .tile > *  > * {
      background-color: cadetblue;
      color: white;
    }
  `)}
  />
  <Media size="small">
    <EzCard>
      <EzLayout layout="tile" className="tile" columns={{base: 1, medium: 3}}>
        <div>Content</div>
        <div>Content</div>
        <div>Content</div>
        <div>Content</div>
        <div>Content</div>
      </EzLayout>
    </EzCard>
  </Media>
</div>
```

### Resets margin correctly

```jsx
<EzCard>
  <EzCardSection>
    <EzLayout layout={{base: 'cluster', medium: 'basic'}}>
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
    </EzLayout>
  </EzCardSection>
  <EzCardSection>
    <EzLayout layout={{base: 'basic', medium: 'cluster'}}>
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
      <Placeholder height={48} width={48} />
    </EzLayout>
  </EzCardSection>
</EzCard>
```

### Applies new margin correctly to elements with margin rest

```jsx
<EzPage>
  <EzCard>
    <EzCardSection>
      <EzLayout layout="stack">
        <EzHeading size={3}>Growth from your site & ours</EzHeading>
        <p>We've spent years solving the complexities of catering orders.</p>
      </EzLayout>
    </EzCardSection>
  </EzCard>
</EzPage>
```
