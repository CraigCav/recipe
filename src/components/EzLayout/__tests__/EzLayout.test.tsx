import React from 'react';
import {axe} from 'jest-axe';
import {visualSnapshots} from 'sosia';
import {Global, css} from '@emotion/core';
import markdown from '../EzLayout.md';
import regressionTests from './EzLayout.test.md';
import EzLayout from '../EzLayout';
import {EzCard, EzCardSection, EzHeading} from '../../index';
import {renderToHtml} from '../../../jest-globals';
import Media from '../../EzField/Media';
import Placeholder from '../../../../doc-site/src/components/Placeholder';

const scope = {EzLayout, EzCard, EzCardSection, EzHeading, Media, css, Global, Placeholder};

describe('EzLayout', () => {
  visualSnapshots({markdown, scope});
  visualSnapshots({markdown: regressionTests, scope});

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(
      <EzLayout layout="basic">
        <div>Content</div>
        <div>Content</div>
        <div>Content</div>
        <div>Content</div>
      </EzLayout>
    );
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });

  it('should pass type checking', () => {
    [
      {
        defaultLayout: (
          <EzLayout>
            <div>Content</div>
            <div>Content</div>
          </EzLayout>
        ),
        basicLayout: (
          <EzLayout layout="basic">
            <div>Content</div>
            <div>Content</div>
          </EzLayout>
        ),
        tileLayout: (
          <EzLayout layout="tile" columns={3}>
            <div>Content</div>
            <div>Content</div>
          </EzLayout>
        ),
        tileLayoutResponsiveColumns: (
          <EzLayout layout="tile" columns={{base: 1, medium: 3}}>
            <div>Content</div>
            <div>Content</div>
          </EzLayout>
        ),
        basicResponsiveLayout: (
          <EzLayout layout={{base: 'basic', medium: 'stack'}}>
            <div>Content</div>
            <div>Content</div>
          </EzLayout>
        ),
        tileResponsiveLayout: (
          <EzLayout layout={{base: 'basic', medium: 'tile'}} columns={3}>
            <div>Content</div>
            <div>Content</div>
          </EzLayout>
        ),
      },
    ].forEach(() => {});

    expect.assertions(0);
  });
});
