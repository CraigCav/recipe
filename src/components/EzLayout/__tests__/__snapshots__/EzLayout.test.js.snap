// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzLayout should render with a split layout 1`] = `
.recipe-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.recipe-0 > *:not(:first-child) {
  margin-left: 0.75rem;
}

<div
  className="recipe-0 recipe-1"
>
  <div>
    Content
  </div>
  <div>
    Content
  </div>
</div>
`;

exports[`EzLayout should render with default styles 1`] = `
.recipe-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.recipe-0 > *:not(:first-child) {
  margin-left: 0.75rem;
}

<div
  className="recipe-0 recipe-1"
>
  <div>
    Content
  </div>
  <div>
    Content
  </div>
</div>
`;

exports[`EzLayout should render with equally sized content 1`] = `
.recipe-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.recipe-0 > * {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.recipe-0 > *:not(:first-child) {
  margin-left: 0.75rem;
}

<div
  className="recipe-0 recipe-1"
>
  <div>
    Content
  </div>
  <div>
    Content
  </div>
</div>
`;

exports[`EzLayout should render with right aligned content 1`] = `
.recipe-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.recipe-0 > *:not(:first-child) {
  margin-left: 0.75rem;
}

<div
  className="recipe-0 recipe-1"
>
  <div>
    Content
  </div>
  <div>
    Content
  </div>
</div>
`;
